test/python/compiler/test_transpiler.py

```diff
@@ -1617,3 +1617,31 @@ def test_custom_multiple_circuits(self):
         self.assertEqual(len(transpiled), 2)
         self.assertEqual(transpiled[0], expected)
         self.assertEqual(transpiled[1], expected)
+
+
+@ddt
+class TestTranspileParallel(QiskitTestCase):
+    """Test transpile() in parallel."""
+
+    def setUp(self):
+        super().setUp()
+        original_value = os.getenv("QISKIT_PARALLEL", None)
+
+        def restore_parallel():
+            os.environ["QISKIT_PARALLEL"] = original_value
+
+        self.addCleanup(restore_parallel)
+        del os.environ["QISKIT_PARALLEL"]
```

Comment:
This implicitly assumes that `QISKIT_PARALLEL` is in the environment, which might not be the case for people calling `stestr` directly.  We only need the cleanup and the `del` if `original_value is not None` on the first call, I think.

Reply:
I wonder if we could just use `unittest.mock.patch` as a class decorator to temporarily update `os.environ`?

https://docs.python.org/3/library/unittest.mock.html#patch-dict

Reply:
`patch.dict` is a good use case for this, I'll modify it now

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/8952/commits/949703cf31b9133f211ba612626b7bd0c942d399